### ISSUE TITLE
Initiate a Nightmare instance without cookies/cache from old sessions

### ISSUE THREAD

**Issue by alfonsomga** [segmentio/nightmare#345]
I'm trying to figure out how I can start a new and clean Nightmare instance (without cookies or cache from an old session).

In my below example the bot basically navigates to ``m.facebook.com`` types an ``email``and a ``password`` then it clicks on the ``login`` button, the first time the bot did all this correctly but second time I ran this script the bot was already logged and failed, how can I avoid this?

```javascript
	new Nightmare({show: true})
    	.useragent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.80 Safari/537.36')
        .goto('https://m.facebook.com')
        .wait()
        .type('input[placeholder="Email or Phone"]', email)
        .type('input[placeholder="Password"]', password)
        .click('button[name="login"]')
        .wait(1000)
        .wait()
        .exists('div[id="m_home_notice"]')
        .run(function(err, nightmare) {
            if (err) return console.log(err);
            console.log('Done!');
        });
```

**Comment by alfonsomga** [comment]
Ping top 3 contributors @reinpk @matthewmueller @johntitus

**Comment by alfonsomga** [comment]
This issue could be solved with #338

**Comment by bjrmatos** [comment]
@alfonsomga seems like the `web-preferences.partition` option of `BrowserWindow` is what you need

**Comment by rosshinkley** [comment]
@alfonsomga I've hit this too, and as a note for others who are headscratching (and me in the future), the @bjrmatos link has the options as being dash separated instead of camel-cased.  This is because the documentation that is linked to in the Nightmare documentation is on the Electron master branch, and as such, is unreleased (see [here](https://github.com/atom/electron/issues/3431)).  The options for the version Nightmare is currently on are dash-separated.

A quickly hacked together example:

```javascript
var nightmare = new Nightmare({
  show:true,
  skipTaskbar:true,
  webPreferences:{
    partition: 'persist:foo'
  }
})
```
... doesn't do what you'd expect: this should start a new electron window with no taskbar icon and a partition called "foo" (which you could replace with a unique identifier to get the behavior you want, I used 'foo' for brevity).

However,  if you use the dash-separated options:

```javascript
var nightmare = new Nightmare({
  show:true,
  'skip-taskbar':true,
  'web-preferences':{
    partition: 'persist:foo'
  }
})
```
... will start you with a fresh session with no taskbar icon, as you'd expect.

All that being said, +1 to #338.  That would be handy to have regardless.

Maybe the Nightmare docs should link to the versioned documentation instead of the master branch.  Thoughts?

**Comment by bjrmatos** [comment]
> Nightmare docs should link to the versioned documentation

:+1: 

#338 would be handy for some use cases :) :+1:

**Comment by alfonsomga** [comment]
Thanks @bjrmatos @rosshinkley.

**Comment by alfonsomga** [comment]
@rosshinkley +1, when I added ``'web-preferences':{partition: 'persist:foo123'}`` some functions like ``.type()``, ``.click()`` stopped working.

**Comment by rosshinkley** [comment]
@alfonsomga After some tinkering, I figured out what is happening.  The `defaults` library - which is used to merge passed options with default options - does not merge deeply.  When we were passing in options to `web-preferences`, the existing options - the parts that load most of the sugar that allows Nightmare to work - were getting overwritten.

I've gone ahead and submitted a fix, pending approval (#348).

**Comment by alfonsomga** [comment]
@rosshinkley :+1:
